flocker/control/test/test_persistence.py

```diff
@@ -249,3 +250,24 @@ def test_complex_keys(self):
                                manifestations={}, paths={},
                                devices={uuid4(): FilePath(b"/tmp")})
         self.assertEqual(node_state, wire_decode(wire_encode(node_state)))
+
+
+class ConfigurationMigrationTests(SynchronousTestCase):
+    """
+    Tests for ``_ConfigurationMigration``.
+    """
+    configurations_dir = FilePath(__file__).sibling('configurations')
+
+    def test_v1_v2_configuration(self):
+        """
+        A V1 JSON configuration blob is transformed to a V2 configuration
+        blob, with the result validating when loaded.
+        """
+        v1_json = self.configurations_dir.child(
+            b"configuration_v1.json").getContent()
+        v2_json = migrate_configuration(1, 2, v1_json)
+        v2_config = wire_decode(v2_json)
```

Comment:
Still can't use wire_decode because eventually it's going to be the v3 wire_decode. Possibly you need more of a "test_v1_to_latest_configuration" whose assertions may occasionally need to be updated, but ideally not the rest of it. and test a more complex object... or maybe i'm thinking of this wrong. @exarkun how did the axiom tests deal with this?